src/audio/audio.hpp

```diff
@@ -151,15 +162,18 @@ class PipeAudio : public Audio
 	/// The sink to which audio data is sent.
 	std::unique_ptr<AudioSink> sink;
 
-	/// Whether the current file has stopped decoding.
-	bool file_ended;
-
 	/// The current decoded frame.
 	AudioSource::DecodeVector frame;
 
 	/// The current position in the current decoded frame.
 	AudioSource::DecodeVector::iterator frame_iterator;
 
+	/**
+	 * A map of response sink pointers to their last emitted position, in
+	 * seconds.
+	 */
+	std::map<const ResponseSink *, std::uint64_t> last_times;
```

Comment:
something something ownership? unique_ptr or at least a comment as to what actually owns these